packages/nitro-protocol/contracts/ninja-nitro/SingleChannelAdjudicator.sol

```diff
@@ -461,13 +471,16 @@ contract SingleChannelAdjudicator is
         bytes32 destination,
         uint256 amount
     ) internal {
+        address payable recipient = _isExternalDestination(destination)
+            ? _bytes32ToAddress(destination)
+            : AdjudicatorFactory(adjudicatorFactoryAddress).getChannelAddress(destination);
         if (assetHolderAddress == address(0)) {
-            (bool success, ) = _bytes32ToAddress(destination).call{value: amount}(''); //solhint-disable-line avoid-low-level-calls
+            (bool success, ) = recipient.call{value: amount}(''); //solhint-disable-line avoid-low-level-calls
```

Comment:
Why would we call the fallback function instead of sending money directly?

Is that an explicit choice to allow the destination to execute some logic, when it is not an EOA?

Reply:
This is how we make ETH payments currently, on the advice of our auditors. It is [their recommended way]( https://consensys.net/diligence/blog/2019/09/stop-using-soliditys-transfer-now/?__hstc=753710.c72a59d3c5494b876d4985868ce5af84.1609768438752.1618310831104.1618496184700.9&__hssc=753710.1.1618496184700&__hsfp=1872296574) of sending ETH.

The destination contract may emit some events or other logic, yes.